sql/core/src/main/scala/org/apache/spark/sql/execution/objects.scala

```diff
@@ -394,7 +394,11 @@ case class FlatMapGroupsInRExec(
   override def producedAttributes: AttributeSet = AttributeSet(outputObjAttr)
 
   override def requiredChildDistribution: Seq[Distribution] =
-    ClusteredDistribution(groupingAttributes) :: Nil
+    if (groupingAttributes.isEmpty) {
+      AllTuples :: Nil
```

Comment:
should empty == all?

Reply:
You mean empty grouping attributes == all tuples? Yeah, I think no grouping attributes means all tuples are in the one group.